sdk/core/azure-core/src/main/java/com/azure/core/http/policy/DefaultRedirectStrategy.java

```diff
@@ -88,17 +88,19 @@ public DefaultRedirectStrategy(int maxAttempts, String locationHeader, Set<HttpM
     public boolean shouldAttemptRedirect(HttpPipelineCallContext context,
                                          HttpResponse httpResponse, int tryCount,
                                          Set<String> attemptedRedirectUrls) {
-        String redirectUrl = tryGetRedirectHeader(httpResponse.getHeaders(), getLocationHeader());
 
-        if (isValidRedirectCount(tryCount)
-            && redirectUrl != null
-            && !alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)
-            && isValidRedirectStatusCode(httpResponse.getStatusCode())
+        if (isValidRedirectStatusCode(httpResponse.getStatusCode())
+            && isValidRedirectCount(tryCount)
             && isAllowedRedirectMethod(httpResponse.getRequest().getHttpMethod())) {
-            logger.verbose("[Redirecting] Try count: {}, Attempted Redirect URLs: {}", tryCount,
-                attemptedRedirectUrls.toString());
-            attemptedRedirectUrls.add(redirectUrl);
-            return true;
+            String redirectUrl = tryGetRedirectHeader(httpResponse.getHeaders(), getLocationHeader());
+            if (redirectUrl != null && !alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)) {
+                logger.verbose("[Redirecting] Try count: {}, Attempted Redirect URLs: {}", tryCount,
+                    attemptedRedirectUrls.toString());
```

Comment:
nit: Given that we aren't guaranteed to be able to log at `INFO` we should use the `ClientLogger.log(LogLevel, Supplier<String>)` log method that will defer the `toString` call here.